src/datasets/arrow_dataset.py

```diff
@@ -4393,7 +4394,9 @@ def add_column(self, name: str, column: Union[list, np.array], new_fingerprint:
         info = dataset.info.copy()
         info.features.update(Features.from_arrow_schema(column_table.schema))
         table = update_metadata_with_features(table, info.features)
-        return Dataset(table, info=info, split=self.split, indices_table=None, fingerprint=new_fingerprint)
+        return Dataset(
```

Comment:
Could just do it like below without touching the classes, but I don't know this looked better.
```
new_dataset= Dataset(table, info=info, split=self.split, indices_table=None, fingerprint=new_fingerprint)
        new_dataset._indexes = self._indexes
        return new_dataset
```